src/engine.rs

```diff
@@ -393,7 +391,7 @@ where
 }
 fn twice<F1, F2>(f1: F1, f2: F2) -> Box<OpTwice<F1, F2>>
 where
-    F1: FnOnce(&mut StackDrive),
+    F1: FnOnce(&mut StackDrive) -> Option<()>,
```

Comment:
Could this be `bool` instead? Or are you eventually going to replace the `()` with a type with more data?